csrc/device_lower/validation.cpp

```diff
@@ -649,7 +649,7 @@ class ExprValidator : public OptOutDispatch {
     IterDomain* thread_z = nullptr;
     IterDomain* block_z = nullptr;
 
-    for (const auto& loop_id : block_scaling_factor->getLoopDomain()) {
+    for (const auto& loop_id : quantized_output->getLoopDomain()) {
```

Comment:
We should have been using the quantized output as we schedule and generate loops based on this output.